miio/interfaces/vacuum.py

```diff
@@ -0,0 +1,32 @@
+from typing import Dict, Protocol, runtime_checkable
+
+from miio import DeviceStatus
+
+FanspeedPresets = Dict[str, int]
+
+
+@runtime_checkable
+class Vacuum(Protocol):
+    def start(self):
+        """Start vacuuming."""
```

Comment:
Generally it is good to specify more details about behavior of the interface methods, format of the arguments and returning value. This helps to keep implementations more consistent.

Reply:
Agreed. I just made this PR to show the idea, but your earlier proposal feels like a better solution than this.